R/lightgbm.R

```diff
@@ -238,6 +239,18 @@ train_lightgbm <- function(x,
     free_raw_data = free_raw_data
   )
 
+  if (save_tree_error) {
+    print("Enabling save_tree_error")
+    if (exists("valids")) {
+      print("valids exists; adding tree_errors to it")
+      valids$tree_errors <- d
+    } else {
```

Comment:
This conditional branch ensures that the valids set used for record evals doesn't accidentally clobber the valids set created for validation above.